src/shared/utils/rangeToInterval.ts

```diff
@@ -67,23 +67,65 @@ const convertToInterval = (upperOrLower: string) => {
   if (timestamp.toTimeString() === 'Invalid Date') {
     throw new Error(`Unknown custom time: ${upperOrLower}`)
   }
-  // TODO -- handle timezones.
+
   return `timestamp '${timestamp.toISOString()}'`
 }
 
 /** Convert duration to postgres interval.
  *     https://www.postgresql.org/docs/15/datatype-datetime.html#DATATYPE-INTERVAL-INPUT
  */
-export const rangeToInterval = (range: TimeRange) => {
+export const rangeToSQLInterval = (range: TimeRange): string => {
   switch (range.type) {
     case 'selectable-duration':
       return range.sql
     case 'custom':
     case 'duration':
-      const upper = convertToInterval(range.lower)
-      const lower = convertToInterval(range.upper)
-      return `time >= ${upper} AND time <= ${lower}`
+      const lower = convertToSQLTimeSyntax(range.lower)
+      const upper = convertToSQLTimeSyntax(range.upper)
+      return `time >= ${lower} AND time <= ${upper}`
     default:
       return DEFAULT_TIME_RANGE.sql
   }
 }
+
+/** Reference:
+ *     https://docs.influxdata.com/influxdb/cloud/query-data/influxql/explore-data/time-and-timezone/#time-syntax
+ */
+const convertToInfluxQLTimeSyntax = (upperOrLower: string): string => {
+  if (upperOrLower === null || upperOrLower === 'now()') {
+    return 'now()'
+  }
+
+  // relative time
+  if (!!upperOrLower.match(durationRegExp)) {
+    const durationIntoPast = upperOrLower.charAt(0) === '-'
```

Comment:
This is more of an 'fyi', but we tried for a while to replace `!!` operators with the `Boolean()` function just because it's clearer. But do whatever works / is more common in the context of these files which may be different. 😅

Reply:
Good to know! Thanks, I think `Boolean()` is clearer. Updated!